roxie/ccd/ccdsnmp.cpp

```diff
@@ -877,30 +783,53 @@ class CQueryStatsAggregator : public CInterface, implements IQueryStatsAggregato
         }
     };
 
-    QueueOf<QueryStatsRecord, false> recent;
-    CIArrayOf<QueryStatsAggregateRecord> aggregated; // stored with most recent first
+    std::deque<QueryStatsRecord> recent;
+    std::deque<QueryStatsAggregateRecord> aggregated; // stored with most recent first
     unsigned expirySeconds;  // time to keep exact info (rather than just aggregated)
     StringAttr queryName;
     SpinLock lock; // MORE: This could be held this for a while.  Is this significant?  Should it be a CriticalSection?
```

Comment:
Not clear what this is protecting - and looks a bit dangerous if it can be held for a while.

Reply:
Yeah, it's all messy. But whether it's really hurting anything I don't know. I will change to a CritSec